### PR TITLE
fix(nuxt): add key to `.client` component placeholders

### LINKED ISSUE #20061: `.client.vue` scoped styles are missing on ssr
### Environment

- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.3.3`
- Nitro Version:    `2.3.2`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `experimental`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-a1d8zg?file=pages%2Findex.vue

### Describe the bug

Similar to what https://github.com/nuxt/framework/pull/8713 resolved a while back but it seems like the issue persists for `.client.vue` files?

Setting `experimental: {
    treeshakeClientOnly: false,
  },` which was the workaround for the mentioned pr does however not fix the issue

### Additional context

`.client` component on ssr:
image

`.client` component on client side navigation
image

The `<client-only>` wrapped component on the other hand works for both ssr and csr: 
image

### Logs

_No response_

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolve https://github.com/nuxt/nuxt/issues/20061

### ❓ Type of change

Hi :wave: this PR fix `.client` components root nodes missing scopeId by simply adding a `key`
- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by antfu** [comment]
@huang-julien can you explain more about why adding a boolean key can help the scopeId?

**Comment by huang-julien** [comment]
it is just to completly reload the render. 
My guess was that since the placeholder is a `div` and the components root node is also a `div`, vue does not set the `v-data` scope Id. We don't have the issue if the `.client` component renders a `Fragment` or any other tags.

**Comment by antfu** [comment]
Maybe we could assign a random(or fixed) key to the placeholder but not the rendered component?
So placeholder with key should already not be treat the same as the rendered component without key. There shouldn't be much difference in practice but I am a bit worried overriding the `key` user might define, or causing the component to mount multiple times etc.

**Comment by huang-julien** [comment]
ah yes it should be obvious :sweat_smile: ....

File: packages/nuxt/src/app/components/client-only.mjs [hunk 1]
```diff
@@ -37,7 +38,7 @@ export function createClientOnly (component) {
           ? createElementVNode(res.type, res.props, res.children, res.patchFlag, res.dynamicProps, res.shapeFlag)
           : h(res)
       } else {
-        return h('div', ctx.$attrs ?? ctx._.attrs)
+        return h('div', mergeProps(ctx.$attrs ?? ctx._.attrs, { key: randomUUID() }))
```
Comment (danielroe) [inline packages/nuxt/src/app/components/client-only.mjs:41]: do we need this to be random, or is assigning any key good enough?

Comment (huang-julien) [reply packages/nuxt/src/app/components/client-only.mjs:41]: I think it should work too, i'll try

**Review by danielroe — APPROVED** [review]

**Pull request merged**